### PR TITLE
FEATURE: Add `DNF_CONTENTDIR` override support

### LINKED ISSUE #118: [FEATURE] Add dnf contentdir override support
**Is your feature request related to a problem? Please describe.**
When testing a new release access to staging repositories is possible during initial install but not in post-install or in chained tests using saved machine images.

**Describe the solution you'd like**
RelEng publishes repositories to a custom directory in the primary download server. openQA has hooks to allow the installation repositories to point at these repos. After install is complete the local repository definitions will point at production repository directories which may/may not exist.

Provide mechanism to allow toggle to staging repositories in `dnf` configuration. Solution must work for all supported releases with staging repositories.

**Describe alternatives you've considered**
openQA has hooks that allow build/test from Fedora koji and/or pungi composes. Since Rocky 9 builds in Peridot don't use koji/pungi an alternate solution is needed.
Rocky 8, still built in koji, will be migrated to Peridot as well so solution using existing hooks for Fedora koji/pungi are not ideal as effort spent there would become deprecated after the move.
A common solution using `dnf` configuration in `/etc/dnf/vars/contentdir` seems most appropriate.

### PR DESCRIPTION
**Pull request by tcooper**
# Description

This pull request adds support for specifying alternate repository location via `DNF_CONTENTDIR` variable for post-install tests, resolves incorrect selection of `_console*` vs. `_graphical*` login for a handful of tests and provides initial support for proper select of a Rocky major release and use of that to select alternate Getting Started UI based on GNOME version differences in Rocky 8 and Rocky 9+.

A small number of needles are added to support paging through the Gnome 4 Welcome Tour (no code yet to implement/use those needles).

This PR resolves #118  

# How Has This Been Tested?

```bash
openqa-cli api -X POST isos ISO=Rocky-9.0-RC1.1-x86_64-dvd.iso   ARCH=x86_64 DISTRI=rocky FLAVOR=package-set VERSION=9.0 BUILD=-9.0-$(git branch --show-current)-$(date +%Y%m%d.%H%M%S).0 DNF_CONTENTDIR=true

openqa-cli api -X POST isos ISO=Rocky-9.0-RC1.1-x86_64-boot.iso  ARCH=x86_64 DISTRI=rocky FLAVOR=boot-iso VERSION=9.0 BUILD=-9.0-$(git branch --show-current)-$(date +%Y%m%d.%H%M%S).0 GRUB="inst.repo=https://dl.rockylinux.org/stg/rocky/9/BaseOS/x86_64/os/ inst.addrepo=AppStream,https://dl.rockylinux.org/stg/rocky/9/AppStream/x86_64/os/" DNF_CONTENTDIR=true
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tcooper** [comment]
Dang it... don't know where my GPG signing config went. I'll force push signed commits to my branch later today.

**Comment by akatch** [comment]
Works to this point. I get failures on `_graphical_wait_login` with the Welcome Tour as previously.

File: templates.fif.json [hunk 1]
```diff
@@ -623,83 +624,47 @@
         },
         "install_package_set_minimal": {
             "profiles": {
-                "rocky-boot-iso-x86_64-*-64bit": 10,
-                "rocky-boot-iso-x86_64-*-uefi": 11,
-                "rocky-dvd-iso-x86_64-*-64bit": 11,
-                "rocky-dvd-iso-x86_64-*-uefi": 11,
-                "rocky-minimal-iso-x86_64-*-64bit": 10,
-                "rocky-universal-aarch64-*-aarch64": 30,
-                "rocky-universal-x86_64-*-64bit": 30,
```
Comment (lumarel) [inline templates.fif.json:632]: This removes a lot of the tests which Al recently added, is this on purpose?

**Review by lumarel — APPROVED** [review]
Beside the one question, this PR fixes so many tests, looks good to me! (and all mentioned tests also run completely without error now!)

**Review by akatch — APPROVED** [review]
it gives me great pleasure to approve this beautiful PR

**Pull request merged**